src/full/Agda/Interaction/Library.hs

```diff
@@ -316,6 +345,67 @@ stripCommentLines = concatMap strip . zip [1..] . lines
     strip (i, s) = [ (i, s') | not $ null s' ]
       where s' = trimLineComment s
 
+-- | Returns the path of the @executables@ file which lists the trusted executables Agda knows about.
+--
+--   Note: file may not exist.
+--
+getExecutablesFile
+  :: IO ExecutablesFile
+getExecutablesFile = do
+  agdaDir <- getAgdaAppDir
+  let defaults = map (agdaDir </>) defaultExecutableFiles  -- NB: non-empty list
+  files <- filterM doesFileExist defaults
```

Comment:
This would be inefficient if `defaults` was a long list: checking all files for existence and then only caring for the first such. For long lists, a use of `Agda.Utils.ListT` would enable to do the filtering more lazily.

Maybe a short comment that `defaults` is a very short list prevents code readers like me to stumble over this...

Reply:
See [above](https://github.com/agda/agda/pull/4759#discussion_r443252352).

Reply:
I was just asking for a comment, like `-- NB: defaults is a very short list`.

Reply:
Added.